### ISSUE TITLE
DynamoDBContext ignores DynamoDBTableAttribute

### ISSUE THREAD

**Issue by Samuron** [aws/aws-sdk-net#426]
We are using `DynamoDBContext`  in ASP.NET Core project and despite setting `DynamoDBTableAttribute` to our DTOs the client looks up for a table named exactly like a class, completely ignoring name supplied by the attribute.

**Comment by PavelSafronov** [comment]
What version of the SDK are you using? Can you share your code or the simplest code sample that reproduces this issue?

**Comment by Samuron** [comment]
We are using the 3.2.6-beta version of all SDKs. 
I've prepared a simple sample application using local dynamo:
https://github.com/Samuron/DynamoDBContextIssue

Local version doesn't send very descriptive exceptions, but from AWS version we get something like following:
```
System.AggregateException was unhandled by user code
  HResult=-2146233088
  Message=One or more errors occurred. (Requested resource not found: Table: Pcsconfig1 not found)
  Source=System.Private.CoreLib
  StackTrace:
       at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
       at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
       at System.Threading.Tasks.Task.Wait()
       at AWSCloudformationManager.Repositories.PcsConfigRepository.SaveConfigItem(Pcsconfig1 item)
       at AWSConfigurationManager.IntegrationTests.PcsConfigRepositorySaveAndGetTests.SavedItemShouldBeAsExpected()
  InnerException: 
       ErrorCode=ResourceNotFoundException
       HResult=-2146233088
       Message=Requested resource not found: Table: Pcsconfig1 not found
       RequestId=P0M23525RVL9LMNISR6DJJUCAFVV4KQNSO5AEMVJF66Q9ASUAAJG
       Source=AWSSDK.Core
       StackTrace:
            at Amazon.Runtime.Internal.HttpErrorResponseExceptionHandler.HandleException(IExecutionContext executionContext, HttpErrorResponseException exception)
            at Amazon.Runtime.Internal.ErrorHandler.ProcessException(IExecutionContext executionContext, Exception exception)
            at Amazon.Runtime.Internal.ErrorHandler.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.Internal.CallbackHandler.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.Internal.RetryHandler.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.Internal.CallbackHandler.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.Internal.CallbackHandler.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.Internal.ErrorCallbackHandler.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.Internal.MetricsHandler.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.Internal.RuntimePipeline.InvokeSync(IExecutionContext executionContext)
            at Amazon.Runtime.AmazonServiceClient.Invoke[TRequest,TResponse](TRequest request, IMarshaller`2 marshaller, ResponseUnmarshaller unmarshaller)
            at Amazon.DynamoDBv2.AmazonDynamoDBClient.DescribeTable(DescribeTableRequest request)
            at Amazon.DynamoDBv2.DocumentModel.Table.DescribeTable(String tableName)
            at Amazon.Runtime.Internal.Util.Cache`2.GetValueHelper(TKey key, Boolean& isStaleItem, Func`2 creator)
            at Amazon.DynamoDBv2.DocumentModel.Table.LoadTableInfo()
            at Amazon.DynamoDBv2.DataModel.DynamoDBContext.GetTable(String tableName, DynamoDBFlatConfig flatConfig)
            at Amazon.DynamoDBv2.DataModel.DynamoDBContext.GetTargetTable(ItemStorageConfig storageConfig, DynamoDBFlatConfig flatConfig)
            at Amazon.DynamoDBv2.DataModel.DynamoDBContext.SaveHelper[T](T value, DynamoDBOperationConfig operationConfig, Boolean isAsync)
            at Amazon.DynamoDBv2.DataModel.DynamoDBContext.<>c__DisplayClass98_0`1.<SaveAsync>b__0()
            at Amazon.Runtime.Internal.AsyncRunner.<>c__DisplayClass0_0.<Run>b__0()
            at System.Threading.Tasks.Task`1.InnerInvoke()
            at System.Threading.Tasks.Task.Execute()
       InnerException: 
            HResult=-2146233088
            Message=Exception of type 'Amazon.Runtime.Internal.HttpErrorResponseException' was thrown.
            Source=AWSSDK.Core
            StackTrace:
                 at Amazon.Runtime.HttpWebRequestMessage.GetResponse()
                 at Amazon.Runtime.Internal.HttpHandler`1.InvokeSync(IExecutionContext executionContext)
                 at Amazon.Runtime.Internal.Unmarshaller.InvokeSync(IExecutionContext executionContext)
                 at Amazon.Runtime.Internal.ErrorHandler.InvokeSync(IExecutionContext executionContext)
            InnerException:
```

**Comment by PavelSafronov** [comment]
Thanks for the repo. This is a bug in the beta CoreCLR (3.2.x) version of the SDK, though not an issue with the non-beta 3.1.x codebase. Looking into this.

**Comment by sstevenkang** [comment]
This has been fixed in commit [a3eca2de](https://github.com/aws/aws-sdk-net/commit/a3eca2de83287734d15aea1a5dc95fd323d40baf)
